src/pages/blamefile.js

```diff
@@ -1,20 +1,92 @@
+    const [form, setForm] = useState({ ownername: "CSC4350-TR", reponame: "GitHubGo", filepath: "src/App.js" });
+    const [validate, setValidate] = useState({ ownername: false, reponame: false, filename: true })
+    // const [isUser, setUser] = useState("org")
+    const [fetchData, setFetchData] = useState({ userid: "", branches: new Set([]) })
+    
+    const [blamefileData, setfetchBlameFileData] = useState({ startingLine: 0, endingLine: 0, age: 0, name: "", oid:0, date:new Date().toLocaleString(), total:0, url:"", })
+    const [show, setShow] = useState(false);
+    
+    const [isUser, setUser] = useState("org")
+    const isSelected = (val) => isUser === val;
+    const handleSelectChange = (e) => setUser(e.target.value);
+    
+    const handleInput = (e) => {
+        setForm(i => ({ ...i, [e.target.name]: e.target.value }))
+        setValidate(i => ({ ...i, [e.target.name]: false }))
+    }
+    
+   const handleSubmit = async (e) => {
         e.preventDefault();
+        const data = await validateRepoStats(form.ownername, form.reponame, form.filepath, validate, isUser, fetchData);
+        
+        setErrors(i => ({ ...i, ...data.errors }))
+        setValidate(Object.fromEntries(Object.entries(data.errors).map(i => [i[0], !i[1].status])))
+        
+        setFetchData(data.data)
+        
+        if (!data.result) return
+        setfetchBlameFileData(await fetchBlameFileData(form.ownername, form.reponame, (data.data.branches.has("main")) ? "main" : data.data.branches.entries().next().value[0], data.data.userid))
+        setShow(true);
+        // console.log(repostatsData.datalist.map(i => i.additions))
     }
+    
     return (
         <div className="flex items-center flex-col min-h-screen">
             <form className='flex flex-col gap-3 md:flex-row mt-4' onSubmit={handleSubmit}>
-                <input spellCheck={false} title="Owner Name" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='ownername' placeholder="Owner Name" />
-                <input spellCheck={false} title="Repository Name" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='reponame' placeholder="Repository Name" />
-                <input spellCheck={false} title="File path" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='filepath' placeholder="File Path" />
+                <input spellCheck={false} title="Owner Name" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='ownername' placeholder="CSC4350-TR"/>
+                <input spellCheck={false} title="Repository Name" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='reponame' placeholder="GitHubGo"/>
+                <input spellCheck={false} title="File path" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='filepath' placeholder="src/App.js" />
```

Comment:
```suggestion
<input spellCheck={false} title="Owner Name" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='ownername' placeholder="Owner name"/>
                <input spellCheck={false} title="Repository Name" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='reponame' placeholder="Repository Name"/>
                <input spellCheck={false} title="File path" className="bg-gray-50 p-3 rounded-md focus:outline-gray-200" type={'text'} name='filepath' placeholder="File Path" />
```

The user needs to know what input they should put into. A placeholder is like a label that doesn't store the values inside it; it is only for display purposes when there is no input.